rewrite-java/src/main/java/org/openrewrite/java/marker/PackageOnCompactSourceFile.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.marker;
+
+import lombok.Value;
+import lombok.With;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.marker.Marker;
+
+import java.util.UUID;
+
+@Value
+@With
+public class PackageOnCompactSourceFile implements Marker {
+    UUID id;
+    Space prefix;
+    String packageDefinition;
```

Comment:
This shouldn't be a marker, this is just a `J.Package`. This will be missed by any recipe which wants to read or manipulate package declarations.

Reply:
I actually over engineered this to retain a potential package definition on a compact source file but this actually results in a compile error `java: compact source file should not have package declaration` so I'll remove this construct!